python/tskit/trees.py

```diff
@@ -2451,29 +2451,79 @@ def generate_star(num_leaves, *, span=1, branch_length=1, record_provenance=True
-        tc = tables.TableCollection(sequence_length=span)
-        tc.nodes.set_columns(
-            flags=np.full(num_leaves, NODE_IS_SAMPLE, dtype=np.uint32),
-            time=np.zeros(num_leaves),
+        return combinatorics.generate_star(
+            num_leaves,
+            span=span,
+            branch_length=branch_length,
+            record_provenance=record_provenance,
+        )
+
+    @staticmethod
+    def generate_balanced(
+        num_leaves, *, arity=2, span=1, branch_length=1, record_provenance=True
+    ):
+        """
+        Generate a :class:<Tree> with the specified number of leaves that is maximally
+        balanced. By default, the tree returned is binary, such that for each
+        node that subtends :math:`n` leaves, the left child will subtend
+        :math:`\\floor{n / 2}` leaves and the right child the remainder. Balanced
+        trees with higher arity can also generated using the ``arity`` parameter,
+        where the leaves subtending a node are distributed among its children
+        analogously.
+
+        In the returned tree, the leaf nodes are all at time 0, marked as samples,
+        and labelled 0 to n from left-to-right. Internal node IDs are assigned
+        sequentially from n in a postorder traversal, and the time of an internal
+        node is the maximum time of its children plus the specified ``branch_length``.
+
+        :param int num_leaves: The number of leaf nodes in the returned tree (must be
+            be 2 or greater).
+        :param int arity: The maximum number of children a node can have in the returned
+            tree.
+        :param float span: The span of the tree, and therefore the
+            :attr:`~TreeSequence.sequence_length` of the :attr:`.tree_sequence`
+            property of the returned :class:<Tree>.
+        :param float branch_length: The minimum length of a branch in the tree (see
+            above for details on how internal node times are assigned).
+        :return: A star-shaped tree. Its corresponding :class:`TreeSequence` is available
+            via the :attr:`.tree_sequence` attribute.
+        :rtype: Tree
+        """
+        return combinatorics.generate_balanced(
+            num_leaves,
+            arity=arity,
+            span=span,
+            branch_length=branch_length,
+            record_provenance=record_provenance,
         )
-        root = tc.nodes.add_row(time=branch_length)
-        tc.edges.set_columns(
-            left=np.full(num_leaves, 0),
-            right=np.full(num_leaves, span),
-            parent=np.full(num_leaves, root, dtype=np.int32),
-            child=np.arange(num_leaves, dtype=np.int32),
+
+    @staticmethod
+    def generate_comb(num_leaves, *, span=1, branch_length=1, record_provenance=True):
+        """
+        Generate a :class:<Tree> in which all internal nodes have two children
+        and the left child is a leaf. This is a "comb" or "ladder" phylogeny.
```

Comment:
"comb", "ladder", or "pectinate"

Reply:
Oh, you mean a https://en.wikipedia.org/wiki/Caterpillar_tree ? :P

Reply:
Ah, yes, I've heard that as well, now you come to mention it. I asked a few phylogeneticists and they said the convention was "comb" or (more formally) "pectinate" when the tree is rooted. Caterpillar is nice, though.